src/CMakeLists.txt

```diff
@@ -58,6 +55,10 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
   target_link_libraries(benchmark PRIVATE kstat)
 endif()
 
+if (NOT BUILD_SHARED_LIBS)
+  add_definitions(-DBENCHMARK_STATIC_DEFINE)
```

Comment:
These should be `targed_compile_definitions(benchmark PRIVATE -DBENCHMARK_STATIC_DEFINE)`